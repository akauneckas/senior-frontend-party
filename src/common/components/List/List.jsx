import React from 'react';
import './List.scss';

const List = ({ data, columns }) => {
    return (
        <div className="List">
            <div className="List-row List-row--header">
                {columns.map(column => (<div className="List-cell List-cell--header">{ column.header }</div>))}
            </div>
            <div className="List-content">
                {data.map(item => (
                    <div className="List-row">
                        {columns.map(column => {
                            const value = item[column.accessor];
                            const formattedValue = column.suffix ? `${value} ${column.suffix}` : value;
                            return  <div className="List-cell">{ formattedValue }</div>
                        })}
                    </div>
                ))}
            </div>
        </div>
    )
};

export default List;