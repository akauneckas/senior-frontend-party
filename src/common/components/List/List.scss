.List {
  height: 100%;
  &-content {
    max-height: 100%;
    overflow: scroll;
  }
  &-row {
    display: flex;
    justify-content: space-between;
    border-top: 1px solid #e6e6e6;
    &--header {
      color: #f5f5f5;
      background-color: #f5f5f5;
    }
  }
  &-cell {
    color: #666666;
    padding: 1rem;
    &--header {
      color: #999999;
    }
  }
}
