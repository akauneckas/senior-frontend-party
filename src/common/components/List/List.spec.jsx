import React from 'react';
import renderer from 'react-test-renderer';
import List from './List';

describe('<List />', () => {
    it('should match snapshot', () => {
        const data = [
            {
                id: 1,
                name: 'Foo',
                value: 'foo',
            },
            {
                id: 2,
                name: 'Bar',
                value: 'bar',
            },
            {
                id: 3,
                name: 'Buz',
                value: 'buz',
            },
        ];
        const columns = [
            {
                header: 'Name',
                accessor: 'name',
            },
            {
                header: 'Value',
                accessor: 'value',
                suffix: '$',
            },
        ];
        const component = renderer.create(
            <List data={data} columns={columns} />
        );
        expect(component.toJSON()).toMatchSnapshot();
    });
});
