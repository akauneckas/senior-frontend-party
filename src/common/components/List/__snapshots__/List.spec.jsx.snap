// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<List /> should match snapshot 1`] = `
<div
  className="List"
>
  <div
    className="List-row List-row--header"
  >
    <div
      className="List-cell List-cell--header"
    >
      Name
    </div>
    <div
      className="List-cell List-cell--header"
    >
      Value
    </div>
  </div>
  <div
    className="List-content"
  >
    <div
      className="List-row"
    >
      <div
        className="List-cell"
      >
        Foo
      </div>
      <div
        className="List-cell"
      >
        foo $
      </div>
    </div>
    <div
      className="List-row"
    >
      <div
        className="List-cell"
      >
        Bar
      </div>
      <div
        className="List-cell"
      >
        bar $
      </div>
    </div>
    <div
      className="List-row"
    >
      <div
        className="List-cell"
      >
        Buz
      </div>
      <div
        className="List-cell"
      >
        buz $
      </div>
    </div>
  </div>
</div>
`;
